algorithms/constraint_courses_data.json

```diff
@@ -1,1292 +1,1875 @@
-        },
-        {
-            "id": 77,
-            "course_number": 156361,
-            "name": "עקרונות שיטות הצפנה",
-            "aliases": [],
-            "blocked_by": [
-                {
-                    "id": 3,
-                    "can_be_taken_in_parallel": false
-                },
-                {
-                    "id": 7,
-                    "can_be_taken_in_parallel": false
-                },
-                {
-                    "id": 17,
-                    "can_be_taken_in_parallel": false
-                },
-                {
-                    "id": 23,
-                    "can_be_taken_in_parallel": false
-                },
-                {
-                    "id": 24,
-                    "can_be_taken_in_parallel": false
-                },
-                {
-                    "id": 46,
-                    "can_be_taken_in_parallel": false
-                }
-            ]
-        },
-        {
-            "id": 78,
-            "course_number": 157110,
-            "name": "מערכות משובצות",
-            "aliases": [
-                "אמבדד"
-            ],
-            "blocked_by": [
-                {
-                    "id": 6,
-                    "can_be_taken_in_parallel": false
-                },
-                {
-                    "id": 11,
-                    "can_be_taken_in_parallel": false
-                }
-            ]
-        },
-        {
-            "id": 79,
-            "course_number": 157128,
-            "name": "מערכות הפעלה בגישה מחקרית",
-            "aliases": [],
-            "blocked_by": []
-        },
-        {
-            "id": 80,
-            "course_number": 130502,
-            "name": "מבוא להנדסת חשמל א - למחשבים",
-            "aliases": [],
-            "blocked_by": [
-                {
-                    "id": 11,
-                    "can_be_taken_in_parallel": false
-                }
-            ]
-        },
-        {
-            "id": 81,
-            "course_number": 130562,
-            "name": "מעבדה בהנדסת חשמל א",
-            "aliases": [],
-            "blocked_by": [
-                {
-                    "id": 80,
-                    "can_be_taken_in_parallel": true
-                }
-            ]
-        },
-        {
-            "id": 82,
-            "course_number": 156367,
-            "name": "מיחשוב ענן",
-            "aliases": [
-                "מחשוב ענן"
-            ],
-            "blocked_by": [
-                {
-                    "id": 73,
-                    "can_be_taken_in_parallel": false
-                },
-                {
-                    "id": 22,
-                    "can_be_taken_in_parallel": false
-                }
-            ]
-        }
-    ]
+  "version": 1.1,
+  "_comment": "Output which degrees must take the course (mandatory) and which degrees have builtin option to take.",
+  "courses": [
+    {
+      "id": 1,
+      "course_number": 150000,
```

Comment:
It seems you change the eol style (End of line)
Please keep it in lf instead of crlf.